web-components/src/components/combobox/ComboBox.ts

```diff
@@ -1324,86 +1441,97 @@ export namespace ComboBox {
+                  ${!this.checkForVirtualScroll()
+                    ? repeat(
+                        this.filterOptions(this.trimSpace ? this.inputValue.replace(/\s+/g, "") : this.inputValue),
+                        (option: string | OptionMember) => this.getOptionId(option),
+                        (option: string | OptionMember, optionIndex) => {
+                          if (typeof option !== "string" && this.isOptGroup && option.isLabel === "true") {
+                            return html`
+                              <div
+                                part="group-label"
+                                class="group-label"
+                                role="listbox"
+                                aria-controls="md-combobox-listbox"
+                                aria-label=${option.value}
+                                @click=${(e: MouseEvent) => this.toggleGroupListBox(e, option.value)}
+                                @keydown=${(e: KeyboardEvent) => {
+                                  this.handleGroupLabelKeyDown(e, option);
+                                }}
+                                tabindex="1"
+                                aria-selected=${this.getAriaState(optionIndex)}
+                              >
+                                <span part="group-label">${option.value}</span>
+                                ${this.groupArrowButtonTemplate(option.value)}
+                              </div>
+                            `;
+                          } else if (
+                            !(this.isOptGroup && !this.groupExpandedList.includes(this.getOptionGroupName(option)))
+                          ) {
+                            return html`
+                              <li
+                                id=${this.getOptionId(option)}
+                                title="${this.getOptionValue(option)}"
+                                part="combobox-option"
+                                role="option"
+                                class="md-combobox-option"
+                                aria-label=${this.getOptionValue(option)}
+                                aria-selected=${this.getAriaState(optionIndex)}
+                                tabindex="-1"
+                                @click=${this.handleListClick}
+                                aria-checked=${ifDefined(this.isMulti ? this.isOptionChecked(option) : undefined)}
+                              >
+                                ${this.isMulti
+                                  ? html`
+                                      <span class="select-option">
+                                        <md-icon name="icon-check_14"></md-icon>
+                                      </span>
+                                    `
+                                  : nothing}
+                                <span part="label" class="select-label">
+                                  ${this.isCustomContent
```

Comment:
extract separate component to simplify it

Reply:
Fixed